### PR TITLE
More templates

### PR DESCRIPTION
**Pull request by johnboiles**
Templates for UIViewController and UITableViewCell subclasses that are backed by YOViews. I'd love your feedback on these patterns. Hoping to establish best practices for these common classes in the templates.

### CODE REVIEW

**Comment by gabriel** [comment]
:ship:

File: FileTemplates/YOView.xctemplate/___FILEBASENAME___.m [hunk 1]
```diff
@@ -14,8 +14,14 @@ @interface ___FILEBASENAMEASIDENTIFIER___ ()
 @implementation ___FILEBASENAMEASIDENTIFIER___
 
 - (void)viewInit {
+
+    // Initialize subviews here
+
     __weak typeof(self) weakSelf = self;
```
Comment (gabriel) [inline FileTemplates/YOView.xctemplate/___FILEBASENAME___.m]: You could uses YOSelf weakSelf = self;

Comment (johnboiles) [inline FileTemplates/YOView.xctemplate/___FILEBASENAME___.m]: Yeah totally.

I kinda go back and forth on `YOSelf` vs `__weak typeof(self)`. Here's my thinking: I prefer to keep a consistent pattern across the codebase for weak references, and I'm not going to import `YOLayout.h` just for `YOSelf`. I've created a snippet that autocompletes on `__weak` for the weak reference pattern, so it doesn't really take any extra time.

Comment (johnboiles) [inline FileTemplates/YOView.xctemplate/___FILEBASENAME___.m]: On the flipside, I really like the name `YOSelf` :)

**Pull request merged**